### PR TITLE
feat(ecs-patterns): support runtimePlatform for fargate patterns

### LINKED ISSUE #18462: ECS Patterns: Add `RuntimePlatform ` to L3 Constructs
### Description

Add [this](https://github.com/aws/aws-cdk/issues/17242) `RuntimePlatform` property to ECS Patterns constructs. I'm specifically interested in QueueProcessingFargateService and NetworkLoadBalancedFargateService

### Use Case

I want to run Fargate containers on Graviton instances

### Proposed Solution

Implement the `RuntimePlatform` property recently added [here](https://github.com/aws/aws-cdk/issues/17242).

### Other information

_No response_

### Acknowledge

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by michaeldrey**
feat(aws-ecs-patterns): support runtime platform for fargate patterns

Using the implementation of runtimePlatform from [this previous commit](https://github.com/aws/aws-cdk/issues/17242), I have added this functionality to the L3 constructs for the fargate patterns. The [request](https://github.com/aws/aws-cdk/issues/18462) asked for this feature to be added to `QueueProcessingFargateService` and `NetworkLoadBalancedFargateService`, however I felt it should be added to the rest of the Fargate patterns.

I have added integration tests for each pattern with the runtimePlatform property and have added a section to the readme about how to use it.

closes #18462

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts [hunk 1]
```diff
@@ -92,6 +92,12 @@ export interface ApplicationLoadBalancedFargateServiceProps extends ApplicationL
    * @default - A new security group is created.
    */
   readonly securityGroups?: ISecurityGroup[];
+  /**
+   * The runtime platform of the task definition
+   *
+   * @default - undefined
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts:98]:
```suggestion
   * @default undefined
```

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-multiple-target-groups-fargate-service.ts [hunk 2]
```diff
@@ -82,6 +82,13 @@ export interface ApplicationMultipleTargetGroupsFargateServiceProps extends Appl
    * @default Latest
    */
   readonly platformVersion?: FargatePlatformVersion;
+
+  /**
+   * The runtime platform of the task definition
+   *
+   * @default - undefined
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-multiple-target-groups-fargate-service.ts:89]:
```suggestion
   * @default undefined
```

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts [hunk 3]
```diff
@@ -85,6 +85,12 @@ export interface NetworkLoadBalancedFargateServiceProps extends NetworkLoadBalan
    * @default Latest
    */
   readonly platformVersion?: FargatePlatformVersion;
+  /**
+   * The runtime platform of the task definition
+   *
+   * @default - undefined
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-load-balanced-fargate-service.ts:91]:
```suggestion
   * @default undefined
```

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-multiple-target-groups-fargate-service.ts [hunk 4]
```diff
@@ -82,6 +82,12 @@ export interface NetworkMultipleTargetGroupsFargateServiceProps extends NetworkM
    * @default Latest
    */
   readonly platformVersion?: FargatePlatformVersion;
+  /**
+   * The runtime platform of the task definition
+   *
+   * @default - undefined
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/network-multiple-target-groups-fargate-service.ts:88]:
```suggestion
   * @default undefined
```

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/queue-processing-fargate-service.ts [hunk 5]
```diff
@@ -99,6 +99,12 @@ export interface QueueProcessingFargateServiceProps extends QueueProcessingServi
    * @default false
    */
   readonly assignPublicIp?: boolean;
+  /**
+   * The runtime platform of the task definition
+   *
+   * @default - undefined
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/queue-processing-fargate-service.ts:105]:
```suggestion
   * @default undefined
```

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/scheduled-fargate-task.ts [hunk 6]
```diff
@@ -69,6 +69,12 @@ export interface ScheduledFargateTaskImageOptions extends ScheduledTaskImageProp
    * @default 512
    */
   readonly memoryLimitMiB?: number;
+  /**
+   * The runtime platform of the task definition
+   *
+   * @default - undefined
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/scheduled-fargate-task.ts:75]:
```suggestion
   * @default undefined
```

**Review by comcalvi — CHANGES_REQUESTED** [review]
This property (along with all the others that these props share, like `cpu`) should be extracted into its own level of props that they all inherit, instead of having to just add this prop everywhere. @rix0rrr thoughts on this?

**Comment by michaeldrey** [comment]
@comcalvi That makes sense. I can do that in this PR (imo scope creep this out a bunch), or create a new issue/PR for it.

Looking at the props, aside from `scheduled-fargate-task`, they're all the same with the exception of a few have the additional `securityGroups` prop. Condensing them into a base interface that they inherit will make all props consistent across the services as they each end up calling `FargateService`

`scheduled-fargate-task` already inherits from `scheduled-ecs-task` base so I don't believe that needs to be included (but i could be wrong)

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts [hunk 7]
```diff
@@ -92,6 +92,13 @@ export interface ApplicationLoadBalancedFargateServiceProps extends ApplicationL
    * @default - A new security group is created.
    */
   readonly securityGroups?: ISecurityGroup[];
+
+  /**
+   * The runtime platform of the task definition
+   *
+   * @default undefined If not specified, `operatingSystemFamily` is LINUX and `cpuArchitecture` is X86_64
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts:99]: Do not use the value undefined in describing @default values. This is part of the language specification. Instead, use a dash (-) as the first word and describe the behavior of the infrastructure the construct exhibits when the value is not supplied (See [Contributing Guide, Defaults](https://github.com/aws/aws-cdk/blob/v1-main/docs/DESIGN_GUIDELINES.md#defaults)).

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts [hunk 8]
```diff
@@ -92,6 +92,13 @@ export interface ApplicationLoadBalancedFargateServiceProps extends ApplicationL
    * @default - A new security group is created.
    */
   readonly securityGroups?: ISecurityGroup[];
+
+  /**
+   * The runtime platform of the task definition
+   *
+   * @default - If the property is undefined, `operatingSystemFamily` is LINUX and `cpuArchitecture` is X86_64
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts:99]:
```suggestion
   * @default - `operatingSystemFamily` is LINUX and `cpuArchitecture` is X86_64
```

same for the others.

**Review by comcalvi — CHANGES_REQUESTED** [review]
This would be a good case for a separate PR. Feel free to assign the new PR to me, and please drop a comment on this PR linking the new one. Thanks!

**Comment by TheRealAmazonKendra** [comment]
@michaeldrey I can't quite tell here if the items Calvin wanted extracted out into a separate PR have been done. Can you let me know and also let me know if this is ready for another round of reviews?

**Comment by michaeldrey** [comment]
@TheRealAmazonKendra they haven't yet, ill work on getting a PR created for it this week

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Just assigning this to myself and putting this into changes requested so I get the alert when change have been made

**Comment by michaeldrey** [comment]
New PR opened with this change + the shared props

**Pull request closed without merging**